extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -446,23 +544,65 @@ export class AmpAutocomplete extends AMP.BaseElement {
    * @private
    */
   inputHandler_() {
+    if (
```

Comment:
Reverse condition for cleaner short-circuit:

```js
if (
  this.binding_.shouldAutocomplete(
    this.trigger_,
    dev().assertElement(this.inputElement_)
  )
) {
  return;
}

return this.mutateElement(() => {
  this.clearAllItems_();
});
```

Reply:
I'm confused, the reverse would be:
```
if (
  this.binding_.shouldAutocomplete(
    this.trigger_,
    dev().assertElement(this.inputElement_)
  )
) {
  // If the input is the first entry in the field.
    const firstInteraction =
      this.userInput_.length === 0 && this.inputElement_.value.length === 1;
    this.userInput_ = this.binding_.updateUserInput(
      this.trigger_,
      dev().assertElement(this.inputElement_)
    );

    return this.maybeFetchAndAutocomplete_(firstInteraction);
}

return this.mutateElement(() => {
  this.clearAllItems_();
});
```

Reply:
I'm confused, wouldn't the reverse be: 

```js
if (
  this.binding_.shouldAutocomplete(
    this.trigger_,
    dev().assertElement(this.inputElement_)
  )
) {
  // If the input is the first entry in the field.
  const firstInteraction =
    this.userInput_.length === 0 && this.inputElement_.value.length === 1;
  this.userInput_ = this.binding_.updateUserInput(
    this.trigger_,
    dev().assertElement(this.inputElement_)
  );

  return this.maybeFetchAndAutocomplete_(firstInteraction);
}

return this.mutateElement(() => {
  this.clearAllItems_();
});
```

Reply:
I can't read, you're right. As-is is better.